CrossAppLogin/Back/src/main/kotlin/com/infomaniak/core/crossapplogin/back/BaseCrossAppLoginViewModel.kt

```diff
@@ -40,18 +50,64 @@ import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.transform
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.plus
 import kotlinx.serialization.ExperimentalSerializationApi
+import okhttp3.OkHttpClient
 import com.infomaniak.core.R as RCore
 import com.infomaniak.core.appintegrity.R as RAppIntegrity
 import com.infomaniak.core.network.R as RCoreNetwork
 
+private val apiRepository = object : ApiRepositoryCore() {}
+
+private val baseOkHttpClient by lazy {
+    OkHttpClient.Builder().apply {
+        addCache()
+        addCommonInterceptors()
+    }
+}
+
 abstract class BaseCrossAppLoginViewModel(applicationId: String, clientId: String) : ViewModel() {
+
     private val _availableAccounts = MutableStateFlow(emptyList<ExternalAccount>())
     val availableAccounts: StateFlow<List<ExternalAccount>> = _availableAccounts.asStateFlow()
     val skippedAccountIds = MutableStateFlow(emptySet<Long>())
 
+    val accountsCheckingState: StateFlow<AccountsCheckingState> = _availableAccounts.transform { accounts ->
+        if (accounts.isEmpty()) return@transform
+
+        emit(AccountsCheckingState(status = Ongoing))
+
+        var checkedAccounts = mutableListOf<ExternalAccount>()
+        var checkError: Error = Error.Network
+
+        accounts.forEach { account ->
+            accountCheckStatuses.useElement(account) { statusAsync ->
+                val status = statusAsync.await()
+                when (status) {
+                    Ongoing -> if (checkedAccounts.contains(account).not()) {
+                        checkedAccounts.add(account)
+                        emit(AccountsCheckingState(status = Ongoing, checkedAccounts = { checkedAccounts }))
+                    }
+                    is Error -> if (status is Error.Unknown) checkError = Error.Unknown
+                    WaitingForAccount, Complete -> Unit // Cannot happened, the checkStatuses don't return these values
+                }
+            }
+        }
```

Comment:
It'd be nice to have this work in parallel, especially since this is where all the work starts.

I'll think about how we can do it.